src/libs/actions/PaymentMethods.js

```diff
@@ -107,7 +107,56 @@ function openPaymentsPage() {
         ],
     };
 
-    return API.read('OpenPaymentsPage', {}, onyxData);
+    return API.read('OpenPaymentsPage', {
+        // We're passing this to have the data returned in the right format.
+        // This can be removed when the massageData parameter
+        // is removed from here https://github.com/Expensify/Web-Expensify/blob/main/lib/BankAccountAPI.php#L1064.
+        massageData: true,
+    }, onyxData);
+}
+
+/**
+ *
+ * @param {Number} bankAccountID
+ * @param {Number} fundID
+ * @param {Object} previousPaymentMethod
+ * @param {Object} currentPaymentMethod
+ * @param {Boolean} optimistic
+ * @return {Array}
+ *
+ */
+function getMakeDefaultPaymentOnyxData(bankAccountID, fundID, previousPaymentMethod, currentPaymentMethod, optimistic = true) {
+    return [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.USER_WALLET,
+            value: {
+                walletLinkedAccountID: bankAccountID || fundID,
+                walletLinkedAccountType: bankAccountID ? CONST.PAYMENT_METHODS.BANK_ACCOUNT : CONST.PAYMENT_METHODS.DEBIT_CARD,
+                errors: null,
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: previousPaymentMethod.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT ? ONYXKEYS.BANK_ACCOUNT_LIST : ONYXKEYS.CARD_LIST,
+            value: {
+                [previousPaymentMethod.methodID]: {
+                    ...previousPaymentMethod,
```

Comment:
Do we really need to update all the props of the payment method? The only prop we want to update is `isDefault`, correct?